Generals/Code/GameEngine/Source/GameClient/InGameUI.cpp

```diff
@@ -5454,10 +5466,8 @@ void InGameUI::selectNextIdleWorker( void )
 	{	
 		
 		//If our idle worker is contained by anything, we need to select the container instead.
-		Object *containedBy = selectThisObject->getContainedBy();
-		if( containedBy )
-		{
-			selectThisObject = containedBy;
+		while (selectThisObject->getContainedBy()) {
+			selectThisObject = selectThisObject->getContainedBy();
```

Comment:
I expect this to be obsoleted and removed if the new logic above works correctly.